docs/pallas/quickstart.md

```diff
@@ -0,0 +1,230 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.7
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Pallas Quickstart
+
+Pallas is an extension to JAX that enables writing custom kernels for GPU and TPU. Pallas allows you to use the same JAX functions and APIs but operates at a *lower* level of abstraction.
+
+Specifically, Pallas requires users to think about memory access and how to divide up computations across multiple compute units in a hardware accelerator. On GPUs, Pallas lowers to Triton and on TPUs, Pallas lowers to Mosaic.
+
+Let's dive into some examples.
+
++++
+
+## Hello world in Pallas
+
+```{code-cell} ipython3
+import jax
+from jax.experimental import pallas as pl
+import jax.numpy as jnp
+import numpy as np
+```
+
+We'll first write the "hello world" in Pallas, a kernel that adds two vectors.
+
+```{code-cell} ipython3
+def add_vectors_kernel(x_ref, y_ref, o_ref):
+  x, y = x_ref[...], y_ref[...]
+  o_ref[...] = x + y
+```
+
+**`Ref` types**
+
+Let's dissect this function a bit. Unlike most JAX functions you've probably written, it does not take in `jax.Array`s as inputs and doesn't have any outputs. Instead it takes in *`Ref`* objects as inputs. Note that we also don't have any outputs but we are given an `o_ref`, which corresponds to the desired output.
+
+**Reading from `Ref`s**
+
+In the body, we are first reading from `x_ref` and `y_ref`, indicated by the `[...]` (the ellipsis means we are reading the whole `Ref`; alternatively we also could have used `x_ref[:]`). Reading from a `Ref` like this returns a `jax.Array`.
```

Comment:
Any reason to prefer `...` to `:`? I find the `:` syntax much more compact and readable.

Reply:
`...` works even with ()-shaped `Ref`s whereas `:` does not.